vermouth/tests/gmx/test_itp.py

```diff
@@ -74,3 +78,127 @@ def test_header(tmpdir, dummy_molecule):
     with open(str(outpath)) as infile:
         for line, expected_line in zip(infile, expected):
             assert line == expected_line
+
+
+@pytest.mark.parametrize('pre_meta', (True, False))
+@pytest.mark.parametrize('post_meta', (True, False))
+def test_pre_post_section_lines(pre_meta, post_meta):
+    """
+    :func:`vermouth.gmx.itp.write_molecule_itp` writes pre and post section lines.
+
+    If `pre_meta` or `post_meta` is `True`, then "pre_section_lines" or
+    "post_section_lines", respectively, is read from `molecule.meta`, otherwise
+    it is passed as an argument.
+    """
+    molecule = Molecule(nrexcl=1)
+    molecule.add_nodes_from((
+        (0, {
+            'atype': 'Q5', 'resid': 1, 'resname': 'VAL', 'atomname': 'BB',
+            'charge_group': 1, 'charge': 1,
+        }),
+        (1, {
+            'atype': 'Q5', 'resid': 1, 'resname': 'VAL', 'atomname': 'BB',
+            'charge_group': 1, 'charge': 1,
+        }),
+        (2, {
+            'atype': 'Q5', 'resid': 1, 'resname': 'VAL', 'atomname': 'BB',
+            'charge_group': 1, 'charge': 1,
+        }),
+    ))
+    # The molecule has 6 interaction sections. The sections are names
+    # "interaction_<joint>_<part>", where <joint> can be "with" or "only", and
+    # <part> can be "pre", "post", or "pre_post". The <part> indicates what
+    # pre_section_lines and post_section_lines are defined for the section.
+    # When <joint> in "with", then the section contains 2 interactions in
+    # addition to the pre- and post- lines. When <joint> is "only", then the
+    # section only has the lines defined, and no interactions.
```

Comment:
the lines? I guess pre/post lines?
And could/should this be pytest parametrized?

Reply:
Fixed the phrasing. The test could be parametrized, it was my initial idea. Though, a brief trial with that way lead to a less clear test.

Reply:
I find the test fairly mind-boggling as is, but ok.